ui/jstests/test_manage_taxonomies.jsx

```diff
@@ -91,4 +378,411 @@ define(['QUnit', 'jquery', 'setup_manage_taxonomies', 'reactaddons',
+    function(assert) {
+      assert.ok(AddVocabulary, "class object not found");
+      var vocabularies = [
+        {
+          "vocabulary": vocabulary,
+          "terms": vocabulary.terms
+        }
+      ];
+      var learningResourceTypes = ['course', 'chapter', 'sequential',
+        'vertical', 'html', 'video', 'discussion', 'problem'];
+      var saveVocabularyResponse;
+      var done = assert.async();
+      var updateParent = function(data) {
+        saveVocabularyResponse = data;
+      };
+      var afterMount = function(component) {
+        assert.equal(
+          component.state.name,
+          ''
+        );
+        assert.equal(
+          component.state.description,
+          ''
+        );
+        assert.equal(
+          component.state.vocabularyType,
+          'm'
+        );
+        assert.equal(
+          component.state.learningResourceTypes.length,
+          0
+        );
+        /* assertAddVocabularyCommon submits the form, checking state
+           is clear in waitForAjax */
+        var formNode = assertAddVocabularyCommon(assert, component);
+        waitForAjax(1, function() {
+          assert.equal(
+            saveVocabularyResponse.id,
+            vocabulary.id
+          );
+          //testing state is reset
+          assert.equal(
+            component.state.name,
+            ''
+          );
+          assert.equal(
+            component.state.description,
+            ''
+          );
+          assert.equal(
+            component.state.vocabularyType,
+            'm'
+          );
+          assert.equal(
+            component.state.learningResourceTypes.length,
+            0
+          );
+          var inputNodes = React.addons.TestUtils.
+          scryRenderedDOMComponentsWithTag(
+            formNode,
+            'input'
+          );
+          var inputVocabularyName = inputNodes[0];
+          var inputVocabularyDesc =  inputNodes[1];
+          var checkboxCourse =  inputNodes[2];
+          var radioTagStyle = inputNodes[11];
+
+          React.addons.TestUtils.Simulate.change(
+            inputVocabularyName, {target: {value: "TestB"}}
+          );
+          React.addons.TestUtils.Simulate.change(
+            inputVocabularyDesc, {target: {value: "TestB"}}
+          );
+          React.addons.TestUtils.Simulate.change(
+            checkboxCourse,
+            {target: {value: 'course', checked: true}}
+          );
+          radioTagStyle.checked = true;
+          React.addons.TestUtils.Simulate.change(
+              radioTagStyle,
+              {target: {value: 'f', checked: true}}
+            );
+          component.forceUpdate(function() {
+            assert.equal(component.state.name, "TestB");
+            assert.equal(component.state.description, "TestB");
+            assert.equal(component.state.vocabularyType, "f");
+            assert.equal(
+              component.state.learningResourceTypes[0],
+              'course'
+            );
```

Comment:
Move the rest of the code in the test inside this function so that it executes in the correct sequence